detox/src/devices/common/drivers/android/tools/ApkHashUtils.js

```diff
@@ -0,0 +1,36 @@
+const fs = require('fs');
+const path = require('path');
+
+const generateHash = require('../../../../../utils/generateHash');
+
+const { FILE_PATH } = require('./TempFileTransfer');
+
+const saveHashToDevice = async ({ tempFileTransfer, bundleId, deviceId, binaryPath }) => {
+  const hashFilename = _getHashFilename(bundleId);
+  await _createLocalHashFile(hashFilename, binaryPath);
+  await tempFileTransfer.prepareDestinationDir(deviceId);
+  await tempFileTransfer.send(deviceId, hashFilename, hashFilename);
+  _deleteLocalHashFile(hashFilename);
+};
+
+const isHashUpdated = async ({ adb, deviceId, bundleId, binaryPath }) => {
```

Comment:
This just doesn't make sense to me... - to have a _utility_ get one of our core dependency objects as an argument.
It's a smell saying this should be a helper that gets injected like all other deps.

The smell itself comes from the fact the giving `adb` as an arg hints a great deal of the internal implementation - hence doesn't encapsulate it all the way. Having a Class that just gets that in the c'tor, limits that to just one function (the c'tor).

Reply:
Also, naming-wise: Would `isHashUpToDate` make more sense?